### PR TITLE
Primal Reversion

### PR DESCRIPTION
**Pull request by Farboo171**
Adds a Primal Reversion quest in the middle of the Delta Episode. This quest:

1. Is very weather related
2. Is set to be late-kalos difficulty, spun out of Delta episode right before the end
3. will allow for catching of Primals in the correct weather after the quest is completed
4. treats primals the same as megas from an Evo perspective

### CODE REVIEW

File: src/scripts/towns/TownList.ts [hunk 1]
```diff
@@ -2577,6 +2577,93 @@ const HoennStoneSalesman2 = new NPC('Stone Salesman', [
 ], {
     requirement: new TemporaryBattleRequirement('Hoenn Stone Salesman'),
 });
+
+const MrStone1 = new NPC('Mr. Stone', [
+    'Hello, $playername$! It\'s good to see you again, but I have some dire news.',
+    'Despite Zinnia having stolen their leaders\ Key Stones, Team Aqua and Team Magma seem to be at itagain, cooking up a scheme involving something called "Primal Reversion".',
```
Comment (Vodovik) [inline src/scripts/towns/TownList.ts:2583]:
```suggestion
    'Despite Zinnia having stolen their leaders\ Key Stones, Team Aqua and Team Magma seem to be at it again, cooking up a scheme involving something called "Primal Reversion".',
```

File: src/scripts/towns/TownList.ts [hunk 2]
```diff
@@ -2577,6 +2577,93 @@ const HoennStoneSalesman2 = new NPC('Stone Salesman', [
...
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Primal Reversion', 1), new QuestLineStepCompletedRequirement('Primal Reversion', 3, GameConstants.AchievementOption.less)]),
+});
+
+const PrimalSteven = new NPC('Steven', [
+    'Hello again, $playername$! I see you\re interested in these murals as well? They seem to be about Groudon and Kyogre.',
+    'So in the ancient world, their primal forms once held this much power…? The super-ancient Pokémon… What terrible strength… But this appearance here… It seems somehow different from Mega Evolution, but somehow related.',
+    '"If you don\'t mind me asking… Do you feel anything in particular when you look up at this wall? A primal world, lost thousands of years in the past… A Legendary Pokémon of tremendous power became humanity\'s greatest threat…',
+    'The terror aroused by that power is clear to see in this ancient artwork. you know who might know more about this? Captain Stern, a sailor who works for my father.',
+    'He\'s usually out to sea, but you can find him in Sea Mauville if the weather is too stormy to sail.',
+], {image: 'assets/images/gymLeaders/Steven.png',
```
Comment (Vodovik) [inline src/scripts/towns/TownList.ts:2608]:
```suggestion
], {image: 'assets/images/gymLeaders/steven.png',
```

File: src/scripts/towns/TownList.ts [hunk 3]
```diff
@@ -2577,6 +2577,93 @@ const HoennStoneSalesman2 = new NPC('Stone Salesman', [
...
+
+const PrimalMural2 = new NPC('Kyogre Mural', [
+    '<i>A large mural depicting  some sort of oceanic whale-like Pokémon. It seems to be engaged in a battle.</i>',
+    '<i>You see someone else looking at the mural: Steven Stone!</i>',
+], {image: 'assets/images/npcs/other/KyogreMural.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Primal Reversion', 1), new QuestLineStepCompletedRequirement('Primal Reversion', 3, GameConstants.AchievementOption.less)]),
+});
+
+const PrimalSteven = new NPC('Steven', [
+    'Hello again, $playername$! I see you\re interested in these murals as well? They seem to be about Groudon and Kyogre.',
```
Comment (Vodovik) [inline src/scripts/towns/TownList.ts:2603]:
```suggestion
    'Hello again, $playername$! I see you\'re interested in these murals as well? They seem to be about Groudon and Kyogre.',
```

File: src/scripts/towns/TownList.ts [hunk 4]
```diff
@@ -2577,6 +2577,93 @@ const HoennStoneSalesman2 = new NPC('Stone Salesman', [
...
+    'On the search for mysteries of the sea, eh? Well then I have one whale of a tale for you! But are yeh worthy?',
+], {image: 'assets/images/npcs/Sailor.png',
+    requirement: new MultiRequirement([new WeatherRequirement([WeatherType.Thunderstorm]), new QuestLineStepCompletedRequirement('Primal Reversion', 3), new QuestLineStepCompletedRequirement('Primal Reversion', 5, GameConstants.AchievementOption.less)]),
+});
+
+const Stern2 = new NPC('Captain Stern', [
+    'I see you can be trusted with the mysteries of the sea!',
+    'I have been seeing a lot of strange activity around the Seafloor Cavern recently. Whirlpools, thunderstorms, and all nature of strange beasts.',
+    'There\'s an old submarine around here that I\'ve been working on upgrading to go explore the lowest parts of that cavern, but need some help getting the parts.',
+    'The hull needs some new Metal Coating, the ballast tanks need filled with Mystic Water, and some Heat Rocks are needed for a dive that deep.',
```
Comment (Vodovik) [inline src/scripts/towns/TownList.ts:2623]: need filling, need to be filled ?

Comment (Farboo171) [reply src/scripts/towns/TownList.ts:2623]: Let's assume Captain stern has a mild pittsburgh accent and leave it?

File: src/scripts/quests/QuestLineHelper.ts [hunk 5]
```diff
@@ -1728,6 +1728,100 @@ class QuestLineHelper {
...
+
+        const talkToPrimalArchie = new TalkToNPCQuest(PrimalArchie, 'Talk to Archie.');
+
+        const talkToPrimalMaxie = new TalkToNPCQuest(PrimalMaxie, 'Talk to Maxie.');
+
+        primalReversionQuestLine.addQuest(new MultipleQuestsQuest(
+            [
+                talkToPrimalArchie,
+                talkToPrimalMaxie,
+            ], 'Fink out what the Team Leaders are up to at Mt. Pyre.'));
```
Comment (Vodovik) [inline src/scripts/quests/QuestLineHelper.ts:1808]:
```suggestion
            ], 'Find out what the Team Leaders are up to at Mt. Pyre.'));
```

**Review by Jaaslet — APPROVED** [review]
Looks good. Thanks for the PR

**Pull request merged**